sqs-tester/src/main/java/uk/gov/justice/laa/crime/microservice/sqstester/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.justice.laa.crime.microservice.sqstester;
+
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.util.WebUtils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    // Handle all exceptions generated by the controller.
+    @ExceptionHandler({ MethodArgumentNotValidException.class, HttpMessageNotReadableException.class })
+    public final ResponseEntity<ApiError> handleException(Exception ex, WebRequest request) {
```

Comment:
Since the behaviour for handling `MethodArgumentNotValidException` and `HttpMessageNotReadableException` is a bit different, I would have a separate `@ExceptionHandler `for each to aid readability.

i.e.
```
    @ExceptionHandler( MethodArgumentNotValidException.class)
.....
    @ExceptionHandler(HttpMessageNotReadableException.class)
```

On a side note, there are a significant number of possible exceptions which can be thrown by various libraries, rare failures etc. rather than try and code for all of them, instead probably best to only code for those most likely to be thrown by your service (and any custom sqs-tester exceptions).